src/linearization.jl

```diff
@@ -0,0 +1,197 @@
+    A = get_A(F)
+    B = get_B(F)
+
+    k = get_k(t, linear_model)
+
+    set_A!(linear_model, A, k)
+    set_B!(linear_model, B, k)
+
+    if T
+        d = discrete_dynamics(Q, nonlinear_model, z) - A*x̄ - B*ū
+        set_d!(linear_model, d, k)
+    end
+end
+
+function _linearize_and_discretize!(::Type{Exponential}, ::Val{T}, linear_model::DiscreteLinearModel, nonlinear_model::AbstractModel, z::AbstractKnotPoint) where T
+    ix, iu = z._x, z._u
+    t = z.t
+    dt = z.dt
+    x̄ = z.z[ix]
+    ū = z.z[iu]
+    
+    # dispatch so as not to always using StaticArray implementation?
+    F = DynamicsJacobian(nonlinear_model)
+    jacobian!(F, nonlinear_model, z)
+    A_c = get_A(F)
+    B_c = get_B(F)
+
+    k = get_k(t, linear_model)
+
+    if T
+        d_c = dynamics(nonlinear_model, z) - A_c*x̄ - B_c*ū
+        _discretize!(Exponential, Val(true), linear_model, A_c, B_c, d_c, k, dt)
+    else
+        _discretize!(Exponential, Val(false), linear_model, A_c, B_c, k, dt)
+    end
+
+    nothing
+end
+
+
+"""
+    discretize!(::Type{Q}, discrete_model::DiscreteLinearModel, continuous_model::ContinuousLinearModel; dt=0.0) where {Q<:Explicit}
+
+Discretize `continuous_model` with given integration type. Put results into `discrete_model`. 
+
+Infers discretization timestep using `get_times` for time varying models. 
+
+!!! warning
+    Time invariant models must specify keyword argument for discretization timestep `dt`.
+"""
+function discretize!(::Type{Q}, discrete_model::DiscreteLinearModel, continuous_model::ContinuousLinearModel; dt=0.0) where {Q<:Explicit}
+    @assert is_time_varying(continuous_model) == is_time_varying(discrete_model)
+    @assert is_affine(continuous_model) == is_affine(discrete_model)
+
+    if is_time_varying(continuous_model)
+        @assert all(get_times(continuous_model) .== get_times(discrete_model))
+
+        N = length(get_times(continuous_model))
+        times = get_times(continuous_model)
+
+        for i=1:N-1
+            dt = times[i+1] - times[i]
+
+            _discretize!(Q, is_affine(continuous_model), discrete_model, continuous_model, i, dt)
+        end
+    else
+        _discretize!(Q, is_affine(continuous_model), discrete_model, continuous_model, 1, dt)
+    end
+end
+
+_discretize!(::Type{Q}, ::Val{false}, discrete_model::DiscreteLinearModel, continuous_model::ContinuousLinearModel, k::Integer, dt) where {Q<:Explicit} = 
+    _discretize!(Q, Val(false), discrete_model, get_A(continuous_model, k), get_B(continuous_model, k), k, dt)
+
+_discretize!(::Type{Q}, ::Val{true}, discrete_model::DiscreteLinearModel, continuous_model::ContinuousLinearModel, k::Integer, dt) where {Q<:Explicit} = 
+    _discretize!(Q, Val(true), discrete_model, get_A(continuous_model, k), get_B(continuous_model, k), get_d(continuous_model, k), k, dt)
+
+function _discretize!(::Type{Exponential}, ::Val{true}, discrete_model::DiscreteLinearModel, A::AbstractMatrix, B::AbstractMatrix, d::AbstractVector, k::Integer, dt)
```

Comment:
I feel these discretization schemes are tangential to the ones in `integration.jl`. I'd like to have one place of integration schemes, rather than placing them in different places and without having them be interchangeable. Can we unify the ways these are used? For example, I should be able to use `RK4` with `discrete_jacobian` or `discretize`.